src/Adapter.php

```diff
@@ -92,7 +97,7 @@ public function push(MessageInterface $message): MessageInterface
 
     public function subscribe(callable $handlerCallback): void
     {
-        $this->runExisting($handlerCallback);
+        $this->run($handlerCallback, true, 5); // TWK TODO timeout should not be hard coded
```

Comment:
Do you plan to resolve it before merge?

Reply:
Can I do it after the merge, I believe the solution should be somewhere in the middleware but need to investigate a bit further so maybe that can be a lession for another day after the merge...